bindings/wallet-core/src/c/settings.rs

```diff
@@ -37,14 +37,38 @@ pub enum Discrimination {
     Test,
 }
 
+pub type Epoch = u32;
+pub type Slot = u64;
+
+#[repr(C)]
+pub struct TimeEra {
+    pub epoch_start: Epoch,
+    pub slot_start: Slot,
+    pub slots_per_epoch: u32,
+}
+
+impl From<TimeEra> for chain_time::TimeEra {
+    fn from(te: TimeEra) -> Self {
+        chain_time::TimeEra::new(
+            te.slot_start.into(),
+            chain_time::Epoch(te.epoch_start),
+            te.slots_per_epoch,
+        )
+    }
+}
+
 /// # Safety
 ///
 /// settings_out must point to valid writable memory
 /// block_0_hash is assumed to point to 32 bytes of readable memory
 pub unsafe fn settings_new(
     fees: LinearFee,
     discrimination: Discrimination,
-    block_0_hash: *const u8,
+    block0_hash: *const u8,
+    block0_date: u64,
+    slot_duration: u8,
+    time_era: TimeEra,
+    transaction_max_expiry_epochs: u8,
     settings_out: *mut *mut Settings,
 ) -> Result {
```

Comment:
The lint is right: this function should be refactored. Maybe there should be a C struct with all the parameters, that would be converted to the Rust struct by the method.